
== Improper Check or Handling of Exceptional Conditions

=== Policy Details

[width=45%]
[cols="1,1"]
|=== 
|Prisma Cloud Policy ID 
| ed492272-36b2-4ed7-970e-d16504d78f8e

|Checkov ID 
|CKV3_SAST_100

|Severity
|LOW

|Subtype
|Build

|Language
|python

|CWEs
|https://cwe.mitre.org/data/definitions/703.html[CWE-703: Improper Check or Handling of Exceptional Conditions]


|=== 

=== Description

This policy detects instances where there is improper handling of exceptional conditions in Python code. It checks for the use of a try-except block with a generic exception type and a pass statement.

Vulnerable code example:

[source,python]
----
try:
  # Any code that may raise an exception
except Exception:
  pass
----

The above code is vulnerable because it catches all types of exceptions without any specific handling. This can lead to potential issues such as hiding critical errors or mistakenly ignoring unexpected exceptions.

=== Fix - Buildtime

To fix this issue, it is recommended to either handle specific exceptions or include appropriate error handling code within the except block.

Secure code example:

[source,python]
----
try:
  # Any code that may raise an exception
except ValueError:
  # Handle specific exception
  print("Invalid value error occurred.")
except FileNotFoundError as e:
  # Handle specific exception
  print(f"File not found: {e.filename}")
----

The secure code example demonstrates the use of specific exception types within the except block. By handling specific exceptions, the code can perform appropriate actions based on the type of exception encountered. This ensures more robust and reliable error handling.
    